### PR TITLE
STM32L4xx: Add GPIOx ASCR & BRR registers

### PR DESCRIPTION
**Pull request by MathiasKoch**
Also corrects naming on L4x5 ADC `Mult` to `Dual` field, as per the reference manual, and adds prescaler field under ADC_CRR.

Fixes https://github.com/stm32-rs/stm32-rs/issues/342

### CODE REVIEW

File: devices/common_patches/l4_gpio_ascr_brr.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+          access: read-write
+          bitOffset: 0
+          bitWidth: 16
+    BRR:
+      description: GPIO port bit reset register
+      addressOffset: 0x28
+      fields:
+        BR:
+          description: These bits are write-only. A read to these bits returns the value 0x0000
+          access: read-only
```
Comment (richardeoin) [inline devices/common_patches/l4_gpio_ascr_brr.yaml:18]: Guess this should be `write-only`

**Review by richardeoin — CHANGES_REQUESTED** [review]
Thanks for the PR! Couple of comments:
* The RM describes the BRR / ASCR registers with a series of one-bit fields `BRR0`, `BRR1`, ... and this is the case for all the other GPIO register definitions also. I think this would be preferred instead of a 16-bit field for consistency if nothing else
* BRR is also missing on stm32l4x1, stm32l4x2, stm32l412, stm32l4x3. These are parts that don't have ASCR, so it would make sense to split the patch file and apply the BRR section to stm32l4x1.. also

**Comment by MathiasKoch** [comment]
@richardeoin Thanks for the feedback.

I believe this should address all your points?

**Review by richardeoin — APPROVED** [review]
Perfect, thanks!

bors r+

**Review by richardeoin — APPROVED** [review]
Tidied merge conflict

bors r+

**Comment by richardeoin** [comment]
Not sure why CI failed. Let's retry

bors retry

**Comment by burrbull** [comment]
Enums in common_patches

**Pull request merged**